micro-rdk/src/common/button.rs

```diff
@@ -0,0 +1,84 @@
+use super::{
+    config::ConfigType,
+    generic::DoCommand,
+    status::{Status, StatusError},
+};
+use crate::google;
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+use thiserror::Error;
+
+pub type ButtonType = Arc<Mutex<dyn Button>>;
+
+#[derive(Debug, Error)]
+pub enum ButtonError {
+    #[error("failed to press button")]
+    FailedPress,
+    #[error("test error")]
+    TestError,
+}
+
+pub trait Button: Status + DoCommand {
```

Comment:
can you also have this require `Send`, per the explanation [here](https://viam.atlassian.net/browse/RSDK-9955)?